pkg/streamingpromql/optimize/plan/commonsubexpressionelimination/optimization_pass_test.go

```diff
@@ -631,6 +636,26 @@ func TestOptimizationPass(t *testing.T) {
 			expectedSelectorsEliminated: 2,
 			expectedSelectorsInspected:  5,
 		},
+		// In this case the foo[5m] smoothed provides step data with both a collection of points but also the alternate smoothed head and tail points.
+		// delta() will use the returned points, but rate() will substitute in the smoothed head/tail points to its calculation.
+		// It is important that if the matrix selector is shared between the functions, that the smoothed head/tail alternate points
+		// are not substituted into the points' collection.
+		"duplicate matrix selectors with smoothed head/tail points": {
```

Comment:
This test only confirms that the generated plan is correct, it won't check that the returned results are correct.

Could you please add a test to `pkg/streamingpromql/testdata/ours/common_subexpression_elimination.test` with an expression like this one? That will confirm that the results are correct.